### ISSUE TITLE
Testing & Prompting Phone Support

### ISSUE THREAD

**Issue by SecUpwN** [SecUpwN/Android-IMSI-Catcher-Detector#269]
[![Androids](https://2.bp.blogspot.com/-AhHUyr7N0zs/VD6pyuSnkCI/AAAAAAAAPgo/jNvxczENZv4/s1600/android-banner.png "Is my smartphone supported?")](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/issues/269)

Folks, I know we're currently piling up on Issues, sorry about that. But trust me, this is good since this conveys we're having very clear ideas about how our App shall work and what needs to be addressed.

Since our project has set a high goal to provide broad support for current consumer smartphones out there, it is important to verify to the user of our App is really working as supposed to on their specific phone. Of course, we currently have our [Requirements](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/wiki/Requirements), but I do want to avoid people thinking that our App works on their new smartphone while in fact it does nothing else than showing a green Status Icon.

**Proposal:**

* Run a set of tests after agreeing to the `DISCLAIMER` which verify necessary detection/protection
* Show a prompt that gives exact detection information (maybe automatically create a logcat on SD)
* If a users phone is not supported, tell user to join our GitHub and upload the logged information

Should this Issue be in fact a duplicate of #173 (which was declined [here](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/issues/173#issuecomment-63829974)), let me know.

**Comment by E3V3A** [comment]
Maybe change the title of this one? It's very ambiguous to know what you mean with "Phone Support". That sound like a completely different thing to me...

**Comment by SecUpwN** [comment]
@E3V3A, what is very ambiguous with it? Since we have set our goal to support as many phones as possible, we should add a prompt as proposed here that shows the user if his phone is indeed making use of all detection and protection functions of our App, or if there is stuff missing. How would they know otherwise?

**Comment by E3V3A** [comment]
I agree completely, but I was just saying that  "Phone Support" is usually understood as a place to call for support, and that's not what you meant. In addition we might wanna add that it should be an *initial* event that runs automatically on first start/install and then can be run manually from *about* page or something...

**Comment by SecUpwN** [comment]
@E3V3A, I've updated the title and added your suggestions to the OP. So how to go about this?

**Comment by SecUpwN** [comment]
I have been re-thinking this Issue and would now like @banjaxbanjo and @DimaKoz to have a word in here: Is it even a good idea running a small set of tests when initially installing our app to make sure it works properly? If it does not work and/or is not a good proposal, I might as well close this again.

**Comment by TPS** [comment]
Since you're calling for more general comments, & since I'm a bit more "user"-level than most everyone here (& sorry in advance if my 2¢ is irrelevant =), I think this is a fantastic idea, given excellent heuristics, especially re: my trouble with https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/issues/457 .

**Comment by marvinmarnold** [comment]
Hi sorry for jumping in very late too. I think this is a tremendously good idea. Unfortunately, I'm not very sure how to implement it. Does anybody know how complex it might be?

**Comment by SecUpwN** [comment]
>Does anybody know how complex it might be?

Thanks for jumping in, @marvinmarnold. My initial thought was to basically quickly test the functions of the phone where our app is installed on, all of this could be done during initial setup in #181, I guess. I know that our project seemed "dead" the past few weeks, I'm working hard to catch up again. @DJaeger, are you able to tell us how the things proposed in here could be accomplished without writing a huge code-overload? Not sure if there exist any scripts for tasks like this.

**Comment by marvinmarnold** [comment]
What functions specifically would need to be tested? I'm going to two conferences soon (ARIN 36 & ITU World Telecom) that I want to have AIMSICD running during. I'm not even really sure if my phone is compatible at this point.

**Comment by DJaeger** [comment]
Its a while since I had time to look at the current code, but last time the most of the tests, what is available to the app, were at the places the feature were used.
There is currently no central place, where we check them and remain on these results.
Because of that we also do some tests multiple times, like if we use the GSM or CDMA classes.
We should "outsource" the tests in a feature detection class and use the results in all other classes.
And this class could also be used for this initial check.

**Comment by agilob** [comment]
@DJaeger what do you mean by tests? There are no technical tests in the app https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/tree/development/app/src/androidTest/java/com/SecUpwN/AIMSICD/test

**Comment by DJaeger** [comment]
I mean for example a simply IF condition if the CDMA class of the TelephonyManager is available.
Or some types of switch statements to determine, what feature is available and can be used.

**Comment by SecUpwN** [comment]
@larsgrefer, do you have any good ideas how we can implement a functionality test so that the user of our app knows which functions will work on his specific device? This test should be done during #181.
